src/info/manifest.c

```diff
@@ -381,6 +381,178 @@ manifestNewInternal(void)
+                    strZ(manifestPathPg(target1->name)), strZ(manifestTargetPath(this, target1)));
+            }
+
+            // Check if the link destination path already exists
+            const ManifestLinkCheckItem *const link = lstFind(linkCheck->linkList, &path);
+
+            if (link != NULL)
+            {
+                // If both links are files make sure they don't link to the same file
+                if (target1->file != NULL && link->file != NULL)
+                {
+                    if (strEq(target1->file, link->file))
+                    {
+                        const ManifestTarget *const target2 = manifestTarget(this, link->targetIdx);
+
+                        THROW_FMT(
+                            LinkDestinationError,
+                            "link '%s' (%s/%s) destination is the same file as link '%s' (%s/%s)",
+                            strZ(manifestPathPg(target1->name)), strZ(manifestTargetPath(this, target1)), strZ(target1->file),
+                            strZ(manifestPathPg(target2->name)), strZ(manifestTargetPath(this, target2)), strZ(target2->file));
+                    }
+                }
+                // Else error because one of the links is a path and cannot link to the same path as another file/path link
+                else
+                {
+                    const ManifestTarget *const target2 = manifestTarget(this, link->targetIdx);
+
+                    THROW_FMT(
+                        LinkDestinationError,
+                        "link '%s' (%s) destination is the same directory as link '%s' (%s)",
+                        strZ(manifestPathPg(target1->name)), strZ(manifestTargetPath(this, target1)),
+                        strZ(manifestPathPg(target2->name)), strZ(manifestTargetPath(this, target2)));
+                }
+            }
+            // Else add to the link list and check against other links
+            else
+            {
+                // Add the link destination path and sort
+                lstAdd(linkCheck->linkList, &(ManifestLinkCheckItem){.path = path, .file = target1->file, .targetIdx = targetIdx});
+                lstSort(linkCheck->linkList, sortOrderAsc);
+
+                // Find the path in the sorted list
+                unsigned int linkIdx = lstFindIdx(linkCheck->linkList, &path);
+                ASSERT(linkIdx != LIST_NOT_FOUND);
+
+                // Check the link destination path to be sure it is not a subpath of a prior link destination path. Skip file
+                // links since they are allowed to be in the same path with each other and in the parent path of a linked
+                // destination path.
+                for (unsigned int priorLinkIdx = linkIdx - 1; priorLinkIdx < linkIdx; priorLinkIdx--)
+                {
+                    const ManifestLinkCheckItem *const priorLink = lstGet(linkCheck->linkList, priorLinkIdx);
+
+                    if (strBeginsWith(path, priorLink->path))
```

Comment:
This could still be a file link at this point since `->file`  isn't checked until after this (my understanding is `->file = NULL` means it is a path link) - so wouldn't that be a problem if there are links to different files in the same path? I'm not sure I saw a test for this.

Reply:
There is a test for it at ManifestTest.c line 1981. But the comment certainly doesn't match the code. I think this may be tricky to test due to ordering. I'll need to look into it further, but my brain does not want to cooperate today.

Reply:
Added the logic to skip file links in ef221c6e23f57e22. Luckily this already had tests.

Also add checks when the path link is the *parent* of a subsequent link. Depending on the link ordering this can happen.